app/controllers/tasks_controller.rb

```diff
@@ -1,21 +1,26 @@
 class TasksController < ApplicationController
   before_action :set_task, only: %i[show edit update destroy]
-  before_action :set_project, only: %i[new index create destroy]
+  before_action :set_project
+  before_action -> { authorize! @task }, only: %i[show edit update destroy]
 
   def index
+    @task = Task.new(project: @project)
+    authorize! @task
+
     @tasks = @project.tasks
   end
 
   def new
-    @task = Task.new
+    @task = Task.new(project: @project)
+    authorize! @task
   end
 
   def create
-    @task = Task.new(task_params)
-    @task.deadline_at = 7.days.after
+    @task = Task.new(task_params.merge({ project: @project, deadline_at: 7.days.after }))
```

Comment:
Deadline doesn't update in params. Since the task with deadline has already been completed, this parameter can be removed at all.